enhancements/machine-api/converting-machine-api-to-cluster-api.md

```diff
@@ -0,0 +1,891 @@
+Machine API and Cluster API represent different API group versions, but with similar resources.
+This workflow allows switching management of the infrastructure resources (eg EC2 instances) from the former API, to the new API.
+
+When the cluster admin wishes to migrate a MachineSet from Machine API to Cluster API, the following procedure is required:
+1. Identify the MachineSet to migrate to Cluster API
+1. Use the `oc edit` or a patch command to update the value of the `spec.authoritativeAPI` field to `ClusterAPI`
+1. The migration controller verifies that the `Snychonrized` condition is currently set to `True`, verifying no long standing synchronization errors
+1. The `status.authoritativeAPI` field is updated to `Migrating` by the migration controller
+1. The Machine API controller acknowledges the change and sets the `Paused` condition to `True`
+1. The sync controller ensures the latest changes are synchronised between the old authoritative resource and the new, the `status.synchronizedGeneration` is then updated to the current generation of the old authoritative resource
+1. The migration controller verifies that the move from Machine API to Cluster API is valid by checking that the synchronized generation is up to date
+1. The migration controller updates `status.authoritativeAPI`  to `ClusterAPI`
+1. The Cluster API controller takes over management of the MachineSet going forwards
+
+To migrate back from Cluster API to Machine API, the procedure is the same, however, the value of the field should be set to `MachineAPI`.
+The transitional state will be `Migrating` again but the sync controller will wait for the Cluster API controllers to acknowledge
+the change before allowing the Machine API controllers take over management of the resource.
+In both cases, the combination of the `Migrating` status value and the spec `MachineAPI`/`ClusterAPI` value allows the migration controller
+to determine which direction the intended migration is currently taking.
+
+When the conversion cannot proceed for some reason, for example, a feature is in use in Machine API that is not present in Cluster API, the following will happen:
+1. Cluster admin identifies a MachineSet to migrate to Cluster API
+1. Cluster admin uses `oc edit` to update the value of the `spec.authoritativeAPI` field to `ClusterAPI`
+1. The migration controller determines that the `Synchronized` condition is set to `False`, showing a long standing conversion error
+1. The `status.authoritativeAPI` field is not updated and the `status.synchronizedGeneration` is not updated
+1. The Machine API controllers continue to manage the MachineSet.
+1. Where appropriate, the cluster admin contacts support for additional help/timelines for availability of the missing feature
+
+Long standing syhcnronization errors, where the admin has requested a transition by changing the `authoritativeAPI`, will result in alerts firing to indicate to the
```

Comment:
> alerts firing

Where are these alerts firing to? Kube `Events`? Metrics? `Status` stanzas?

Reply:
I had meant alerts in terms of prometheus alerts, so we will fire metrics based on this and then use prometheus/alertmanager to fire alerts